### PR TITLE
`/datum/surgery/organ_manipulation/Destroy()` calls parent

### LINKED ISSUE #88172: Surgery sometimes(?) fails to initiate after previous surgeries and causes a runtime error
### Client Version:

515.1641

### Issue Summary:

surgeries failed to initiate with surgical drapes after a previous one was supposedly completed, locking an affected mob from any further surgeries

> So I closed up the chest and the head, and I can no longer operate on the chest or head at all. It also broke the surgery terminal too.

VV inspection of the affected mob showed `surgeries = list(1 = null, 2 = ref to a hardware manipulation surgery datum)`
i did not take screenshots sorry. they were holding a screwdriver in their off hand and nothing was happening when clicking with drapes except for generating a runtime error
the surgery datum had `surgery_flags = 74` and `status = 4`

this bug had occurred several rounds in a row with various timing, affecting doctors operating on humans in the medbay

### Round ID:

243293

### Test Merge Information:

- [Adds meowing and purring back for felinids](https://www.github.com/tgstation/tgstation/pull/87156)

### Reproduction Steps:

lots of these
```
[07:13:11] Runtime in code/_onclick/item_attack.dm, line 167: Cannot read null.surgery_flags
proc name: item interaction (/mob/living/item_interaction)
usr: Steel-Serperior/(Erebus Carrsty)
usr.loc: (Robotics Lab (165,99,3))
src: the monkey (978) (/mob/living/carbon/human/species/monkey)
src.loc: the floor (166,99,3) (/turf/open/floor/iron/white)
call stack:
the monkey (978) (/mob/living/carbon/human/species/monkey): item interaction(Erebus Carrsty (/mob/living/carbon/human), the surgical drapes (/obj/item/surgical_drapes), /list (/list))
the monkey (978) (/mob/living/carbon/human/species/monkey): item interaction(Erebus Carrsty (/mob/living/carbon/human), the surgical drapes (/obj/item/surgical_drapes), /list (/list))
the monkey (978) (/mob/living/carbon/human/species/monkey): base item interaction(Erebus Carrsty (/mob/living/carbon/human), the surgical drapes (/obj/item/surgical_drapes), /list (/list))
the surgical drapes (/obj/item/surgical_drapes): melee attack chain(Erebus Carrsty (/mob/living/carbon/human), the monkey (978) (/mob/living/carbon/human/species/monkey), "icon-x=17;icon-y=13;vis-x=20;v...")
Erebus Carrsty (/mob/living/carbon/human): ClickOn(the monkey (978) (/mob/living/carbon/human/species/monkey), "icon-x=17;icon-y=13;vis-x=20;v...")
the monkey (978) (/mob/living/carbon/human/species/monkey): Click(the floor (166,99,3) (/turf/open/floor/iron/white), "mapwindow.map", "icon-x=17;icon-y=13;vis-x=20;v...")
the monkey (978) (/mob/living/carbon/human/species/monkey): Click(the floor (166,99,3) (/turf/open/floor/iron/white), "mapwindow.map", "icon-x=17;icon-y=13;vis-x=20;v...")
/datum/callback/verb_callback (/datum/callback/verb_callback): Invoke()
/datum/callback/verb_callback (/datum/callback/verb_callback): Invoke()
world: push usr(Erebus Carrsty (/mob/living/carbon/human), /datum/callback/verb_callback (/datum/callback/verb_callback))
/datum/callback/verb_callback (/datum/callback/verb_callback): InvokeAsync()
/datum/callback/verb_callback (/datum/callback/verb_callback): InvokeAsync()
Input (/datum/controller/subsystem/verb_manager/input): run verb queue()
Input (/datum/controller/subsystem/verb_manager/input): fire(0)
Input (/datum/controller/subsystem/verb_manager/input): fire(0)
Input (/datum/controller/subsystem/verb_manager/input): fire(0)
Input (/datum/controller/subsystem/verb_manager/input): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0)
```

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

`/datum/surgery/organ_manipulation/Destroy()` now returns `..()` instead of nothing, kinda important

Fixes #88172 , probably

## Changelog

:cl: Melbert
fix: Maybe fixes some organ manipulation bugs
/:cl:

### CODE REVIEW

File: code/modules/surgery/organ_manipulation.dm [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 /datum/surgery/organ_manipulation/Destroy()
 	if(QDELETED(target) || !HAS_TRAIT(target, TRAIT_FISHING_SPOT))
-		return
+		return ..()
```
Comment (LemonInTheDark) [inline code/modules/surgery/organ_manipulation.dm:39]: why does the lint not catch this wtf

Comment (Kapu1178) [reply code/modules/surgery/organ_manipulation.dm:39]: should_call_parent only checks if there is a parent call anywhere in the proc body, it doesnt check every possible path.

Comment (LemonInTheDark) [reply code/modules/surgery/organ_manipulation.dm:39]: disgusting

**Comment by SmArtKar** [comment]
FUCK

**Review by SmArtKar — APPROVED** [review]

**Pull request merged**